### DISCUSSION TITLE
describe problem

### DISCUSSION THREAD

**Discussion by aidec** [lllyasviel/Fooocus#3407] [Q&A] [answered]
Styles will be reset to default options after using describe
Is there any way to make it only generate a description without changing other settings?

for example
**Before using describe**
Check Fooocus V2, Fooocus Enhance, Flat 2d Art....
**After using describe**
It will become the default style of Preset, such as only selecting Fooocus V2

**Comment by mashb1t** [comment, marked as answer]
You can disable this by adjusting the code. Replace

https://github.com/lllyasviel/Fooocus/blob/a5040f62189efe704df217af98d5596c344636f4/webui.py#L1063-L1073

with

```python
def trigger_describe(mode, img): 
     if mode == flags.describe_type_photo: 
         from extras.interrogate import default_interrogator as default_interrogator_photo 
         return default_interrogator_photo(img)
     if mode == flags.describe_type_anime: 
         from extras.wd14tagger import default_interrogator as default_interrogator_anime 
         return default_interrogator_anime(img)
     return mode
  
 describe_btn.click(trigger_describe, inputs=[describe_method, describe_input_image], 
                    outputs=[prompt], show_progress=True, queue=True) 
```

**Reply by aidec** [reply]
thank you

**Reply by mashb1t** [reply]
implemented in https://github.com/lllyasviel/Fooocus/pull/3430
